imagebuildah/stage_executor.go

```diff
@@ -1269,6 +1325,23 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			}
 		}
 
+		// Following step is just built and was not used from
+		// cache so check if --cache-to was specified if yes
+		// then attempt pushing this cache to remote repo and
+		// fail accordingly.
+		//
+		// Or
+		//
+		// Try to push this cache to remote repository only
+		// if cache was present on local storage and not
+		// pulled from remote source while processing this
+		if (!pulledAndUsedCacheImage || cacheID == "") && s.executor.cacheTo != nil {
```

Comment:
Absolutely non-blocking: Consider
```suggestion
		if s.executor.cacheTo != nil && (!pulledAndUsedCacheImage || cacheID == "") {
```
which reads “which feature is this about? `cacheTo` — and in which cases do we need to do something for that feature?”, that seems more natural to me than “in which cases do we need to do something about `cacheTo`? — and do we even care about `cacheTo` at all?”.